client/src/pages/[room]/index.tsx

```diff
@@ -163,6 +164,9 @@ class Room extends React.Component<Props, State> {
 
         let chatMsg = JSON.parse(encoded_string);
         msgManager.pub(msg.Type, chatMsg);
+      } else if (msg.Type === constants.MSG_TREQUEST_CACHE_CHAT) {
```

Comment:
the cache chat should be no different from the normal chat.
So :
- make the received chat go through `TChat` topic.
- Upgrade the normal to receive a list of chats instead of a single message.


By doing this we only have a single topic to handle chat => simpler design.